stdlib/public/core/BridgeObjectiveC.swift

```diff
@@ -85,6 +85,18 @@ public protocol _ObjectiveCBridgeable {
 
 #if _runtime(_ObjC)
 
+@_silgen_name("swift_stdlib_connectNSBaseClasses")
+internal func _connectNSBaseClasses() -> Bool
+
+/*
+ Reparents the SwiftNativeNS*Base classes to be subclasses of their respective
+ Foundation types, or is false if they couldn't be reparented. Must be run
+ in order to bridge Swift Strings, Arrays, Dictionarys, Sets, or Enumerators
+ to ObjC.
+ */
+internal let _bridgeInitializedSuccessfully = _connectNSBaseClasses()
+
```

Comment:
Is there any mechanism or way to enforce this? Requiring maintainers to just remember to bind a `let` variable whenever they bridge seems very error prone and would lead to difficult to understand bugs.

Reply:
It's only needed for the set of classes that we use the reparenting mechanism for, not bridging in general. So it shouldn't need to expand much beyond this, if any.

Reply:
I spent a while brainstorming ways to enforce this but haven't come up with much so far. The problem is that it's an easy property to check, but once it's set, you lose the ability to check if other code paths set it. So we'd need to somehow have tests that exercised each possible future bridging function before bridging anything else.

Reply:
Slightly more practical might be debug preconditions that check the superclass at a funnel point in each class, but I'm not sure there are good funnel points

Reply:
`objectAtIndex:` or `count` might be a slightly less horrible funnel point, at the cost of "if someone calls something *else* implemented on NSArray first it will just crash".

Reply:
> It's only needed for the set of classes that we use the reparenting mechanism for

That's not the main issue with this approach. This patch makes it so that anyone who wants to instantiate a type, such as `__SwiftEmptyNSEnumerator`, has to somehow know the unwritten convention that they need to call `_connectOrphanedFoundationSubclassesIfNeeded()` first, or else there will be obscure corner case bugs discovered in production.

For example, a more structured approach would be for e.g. `__SwiftNativeNSEnumerator`'s initializer to call `_connectOrphanedFoundationSubclassesIfNeeded()`. That way, at least for that type family, it is an enforced invariant rather than an ad-hoc use-site convention. Is there something preventing this?

Reply:
I initially thought moving it to init would require adding an override of init, which would add an extra objc_msgSend to every initialization of these, but it looks like we have an empty passthrough override anyway.

Even better though: Jordan points out in another comment that you can't get to this except via another bridging call, which will do this initialization, so we can just delete the enumerator calls entirely.